library/printoptions.mli

```diff
@@ -0,0 +1,38 @@
+(** Printing options record *)
+
+(* storage of the options is in the current instance of the record *)
+
+type print_options = {
+  printing_allow_match_default_clause : bool;
+  printing_coercions : bool;
+  printing_compact_contexts : bool;
+  printing_existential_instances : bool;
+  printing_factorizable_match_patterns : bool;
+  printing_implicit : bool;
+  printing_implicit_defensive : bool;
+  printing_matching : bool;
+  printing_notations : bool;
+  printing_primitive_projection_compatibility : bool;
+  printing_primitive_projection_parameters : bool;
+  printing_projections : bool;
+  printing_records : bool;
+  printing_synth : bool;
+  printing_universes : bool;
+  printing_wildcard : bool;
+}
+
+(** getter/setter for Printing options *)
+val get_current_options : unit -> print_options
+val set_current_options : print_options -> unit
+
+(** getter/setters for Printing options *)
+val set_printing_all : unit -> unit
+val unset_printing_all : unit -> unit
+val set_printing_defaults : unit -> unit
+
+(** predicate to see if Printing All in effect *)
+val printing_all : unit -> bool
+
+(** run function in context of temporary printing option(s) *)
+val with_printing_options : (unit -> unit) list -> ('a -> 'b) -> 'a -> 'b
+val with_printing_option : (unit -> unit) -> ('a -> 'b) -> 'a -> 'b
```

Comment:
I think `unit -> unit` should be `print_options -> print_options`.
The list version can be gotten by composing functions so I don't really see the point.

Reply:
Good idea.